src/topology/builder.rs

```diff
@@ -21,6 +22,11 @@ use std::{
 };
 use stream_cancel::{StreamExt as StreamCancelExt, Trigger, Tripwire};
 use tokio::time::{timeout, Duration};
+use vector_core::enrichment;
+
+lazy_static! {
+    static ref ENRICHMENT_TABLES: enrichment::TableRegistry = enrichment::TableRegistry::default();
+}
```

Comment:
What's the reason for moving this into a global?

Reply:
When reloading the config any new components that are loaded need to have access to the existing enrichment tables as well as any new ones. The `build_pieces` function doesn't have access to the old config, and thus the enrichment tables loaded before. 

Moving it to a global means that at all times we have access to only a single set of loaded tables. All clones of the enrichment tables will ultimately point back to this one instance.

I would be more than happy to find an alternative to this and would like to follow this up. I couldn't find a straightforward way. I think it should be possible to copy the current tables in via the `ConfigDiff` struct, but I was hesitant to make significant code changes to this area at this stage of the game.